src/interface/TWElrondTransactionFactory.cpp

```diff
@@ -0,0 +1,109 @@
+
+template <typename ProtoMessage>
+TWData *_Nonnull marshalProto(const ProtoMessage& message);
+std::string unmarshalString(const TWString *_Nonnull value);
+Address unmarshalAddress(const TWString *_Nonnull address);
+uint256_t unmarshalBigUInt(const TWString *_Nonnull value);
+
+struct TWElrondTransactionFactory *_Nonnull TWElrondTransactionFactoryCreate(struct TWElrondNetworkConfig *_Nonnull networkConfig) {
+    auto factory = TransactionFactory(networkConfig->impl);
+    return new TWElrondTransactionFactory{ factory };
+}
+
+void TWElrondTransactionFactoryDelete(struct TWElrondTransactionFactory *_Nonnull self) {
+    assert(self != nullptr);
+    delete self;
+}
+
+TWData *_Nonnull TWElrondTransactionFactoryCreateEGLDTransfer(
+    struct TWElrondTransactionFactory *_Nonnull self,
+    TWString *_Nonnull sender,
+    TWString *_Nonnull receiver,
+    TWString *_Nonnull amount
+) {
+    auto factory = self->impl;
+
+    auto transaction = factory.createEGLDTransfer(
+        unmarshalAddress(sender),
+        unmarshalAddress(receiver),
+        unmarshalBigUInt(amount)
+    );
+
+    return marshalProto(transaction);
+}
+
+TWData *_Nonnull TWElrondTransactionFactoryCreateESDTTransfer(
+    struct TWElrondTransactionFactory *_Nonnull self,
+    TWString *_Nonnull sender,
+    TWString *_Nonnull receiver,
+    TWString *_Nonnull tokenIdentifier, 
+    TWString *_Nonnull amount
+) {
+    auto factory = self->impl;
+
+    auto transaction = factory.createESDTTransfer(
+        unmarshalAddress(sender),
+        unmarshalAddress(receiver),
+        unmarshalString(tokenIdentifier),
+        unmarshalBigUInt(amount)
+    );
+    
+    return marshalProto(transaction);
+}
+
+TWData *_Nonnull TWElrondTransactionFactoryCreateESDTNFTTransfer(
+    struct TWElrondTransactionFactory *_Nonnull self, 
+    TWString *_Nonnull sender,
+    TWString *_Nonnull receiver,
+    TWString *_Nonnull collection,
+    uint64_t nonce,
+    TWString *_Nonnull quantity
+) {
+    auto factory = self->impl;
+
+    auto transaction = factory.createESDTNFTTransfer(
+        unmarshalAddress(sender),
+        unmarshalAddress(receiver),
+        unmarshalString(collection),
+        nonce,
+        unmarshalBigUInt(quantity)
+    );
+    
+    return marshalProto(transaction);
+}
+
+template <typename ProtoMessage>
+TWData *_Nonnull marshalProto(const ProtoMessage& message) {
+    long size = message.ByteSizeLong();
+    Data raw(size);
+    message.SerializeToArray(raw.data(), (int)size);
+    auto result = TWDataCreateWithBytes(raw.data(), size);
+    return result;
+}
+
+std::string unmarshalString(const TWString *_Nonnull value) {
+    return TWStringUTF8Bytes(value);
+}
+
+Address unmarshalAddress(const TWString *_Nonnull address) {
+    Address result;
+    Address::decode(TWStringUTF8Bytes(address), result);
+    return result;
+}
+
+uint256_t unmarshalBigUInt(const TWString *_Nonnull value) {
+    return uint256_t(TWStringUTF8Bytes(value));
```

Comment:
Is there a way to verify if the encoded `value` fits in `uint256_t`?

Reply:
Actually, `boost::multiprecision` contains a checked version of the `uint256`, as well:

```
// Throws: "Unable to allocate sufficient storage for the value of the result: value overflows the maximum allowable magnitude."
boost::multiprecision::checked_uint256_t("115792089237316195423570985008687907853269984665640564039457584007913129639936");
```

While:
```
// The result is: 0
uint256_t("115792089237316195423570985008687907853269984665640564039457584007913129639936");
```

The large number above is `2^256`. `2^256` would indeed fail to be sufficient in case of an extremely large supply of a custom ESDT (the maximum number of decimals is 18). On the other hand, `uint256_t` is the one used throughout TW Core. Perhaps we can propose a TW Core change at a later time, to extend `src/uint256.h` with the checked version, as well?